pkg/kubelet/kubelet_node_status.go

```diff
@@ -60,18 +61,15 @@ func (kl *Kubelet) registerWithApiServer() {
 	if kl.registrationCompleted {
 		return
 	}
-	step := 100 * time.Millisecond
 
+	retries := 0
+	backoff := flowcontrol.NewBackOff(100*time.Millisecond, 10*time.Second)
 	for {
-		time.Sleep(step)
-		step = step * 2
-		if step >= 7*time.Second {
-			step = 7 * time.Second
-		}
-
+		backoff.Next("registerAPIServer", kl.clock.Now())
+		kl.clock.Sleep(backoff.Get("registerAPIServer"))
 		node, err := kl.initialNode()
 		if err != nil {
-			glog.Errorf("Unable to construct v1.Node object for kubelet: %v", err)
+			glog.Errorf("Unable to construct v1.Node object for kubelet: %v retries:", err, retries)
```

Comment:
d% lost. One question, do we really need this `retries` if it's only a counter to show in log?